bio_hansel/subtyper.py

```diff
@@ -437,6 +438,37 @@ def absent_downstream_subtypes(subtype: str, subtypes: pd.Series, scheme_subtype
     return absentees if len(absentees) > 0 else None
 
 
+def missing_nests(st: Subtype, subtype: str, df_positive: pd.DataFrame) -> Subtype:
+    """Find nested subtypes that are missing from the final subtype call
+
+    Args:
+        subtype: Final subtype result
+        positive_subtypes: List of unique positive subtypes found
+    
+    Returns:
+        List of missing hierarchical subtypes or `None` if there are no missing nested hierarchical subtypes.
+    """
+
+    pos_subtypes_str = [x for x in df_positive.subtype.unique()]
+    broken_subs = re.compile(r'^\d(\d*\.+\d+)*$')
+    primary_subtypes = []
```

Comment:
You can make this a `set()` instead of list, then you won't need the `d` Dict below and if you also have `pos_subtypes_str` as a set, instead of looping through all subtypes, you can use set operations to get the items in one set that aren't present in the other (https://docs.python.org/2/library/sets.html#set-objects).